File: 2022/day01.py
import fileinput
from dataclasses import dataclass, field

lines = [line.strip() for line in fileinput.input()]

arr = []
cur = 0
for line in lines:
    if not line:
        arr.append(cur)
        cur = 0
    else:
        cur += int(line)
arr.append(cur)

print(f'🤍 {max(arr):>6}')
print(f'💛 {sum(sorted(arr)[-3:]):>6}')

File: 2022/src/day01.rs
use std::fs;

pub fn main() {
    let contents = fs::read_to_string("1").unwrap();
    let mut sorted: Vec<i32> = contents
        .split("\n\n")
        .map(|set| set.lines().flat_map(|line| line.parse::<i32>()).sum())
        .collect::<Vec<i32>>();
    sorted.sort();
    println!("silver: {}", sorted.last().unwrap());
    println!("gold: {}", sorted.iter().rev().take(3).sum::<i32>());
}
